core/src/main/java/com/datastrato/gravitino/storage/relational/JDBCBackend.java

```diff
@@ -0,0 +1,180 @@
+      NameIdentifier ident, Entity.EntityType entityType, Function<E, E> updater)
+      throws IOException, NoSuchEntityException, AlreadyExistsException {
+    if (entityType == Entity.EntityType.METALAKE) {
+      MetalakePO oldMetalakePO =
+          SessionUtils.getWithoutCommit(
+              MetalakeMetaMapper.class, mapper -> mapper.selectMetalakeMetaByName(ident.name()));
+
+      BaseMetalake oldMetalakeEntity = POConverters.fromMetalakePO(oldMetalakePO);
+      BaseMetalake newMetalakeEntity = (BaseMetalake) updater.apply((E) oldMetalakeEntity);
+      Preconditions.checkArgument(
+          Objects.equals(oldMetalakeEntity.id(), newMetalakeEntity.id()),
+          String.format(
+              "The updated metalake entity id: %s should same with the metalake entity id before: %s",
+              newMetalakeEntity.id(), oldMetalakeEntity.id()));
+
+      Integer updateResult =
+          SessionUtils.doWithCommitAndFetchResult(
+              MetalakeMetaMapper.class,
+              mapper ->
+                  mapper.updateMetalakeMeta(
+                      POConverters.toMetalakePO(newMetalakeEntity), oldMetalakePO));
+      if (updateResult > 0) {
+        return (E) newMetalakeEntity;
+      } else {
+        throw new IOException("Failed to update the entity:" + ident);
+      }
+    } else {
+      throw new IllegalArgumentException(
+          String.format("Unsupported entity type: %s for update operation", entityType));
+    }
+  }
+
+  @Override
+  public <E extends Entity & HasIdentifier> E get(
+      NameIdentifier ident, Entity.EntityType entityType)
+      throws NoSuchEntityException, IOException {
+    if (entityType == Entity.EntityType.METALAKE) {
+      MetalakePO metalakePO =
+          SessionUtils.getWithoutCommit(
+              MetalakeMetaMapper.class, mapper -> mapper.selectMetalakeMetaByName(ident.name()));
+      if (metalakePO == null) {
+        throw new NoSuchEntityException("No such entity:%s", ident.toString());
+      }
+      return (E) POConverters.fromMetalakePO(metalakePO);
+    } else {
+      throw new IllegalArgumentException(
+          String.format("Unsupported entity type: %s for get operation", entityType));
+    }
+  }
+
+  @Override
+  public boolean delete(NameIdentifier ident, Entity.EntityType entityType, boolean cascade) {
+    if (entityType == Entity.EntityType.METALAKE) {
+      Long metalakeId =
+          SessionUtils.getWithoutCommit(
+              MetalakeMetaMapper.class, mapper -> mapper.selectMetalakeIdMetaByName(ident.name()));
+      if (metalakeId != null) {
+        if (cascade) {
+          SessionUtils.doMultipleWithCommit(
+              () ->
+                  SessionUtils.doWithoutCommit(
+                      MetalakeMetaMapper.class,
+                      mapper -> mapper.deleteMetalakeMetaById(metalakeId)),
+              () -> {
+                // TODO We will cascade delete the metadata of sub-resources under the metalake
```

Comment:
I'm not sure if this is the correct behavior.  If there are some resources below the entity, should we reject this operation? An example is we can not drop a database that has some table existing.

Reply:
It's resonable, I take a look at the implementation of KV Store, when sub-resources do exist, an exception is thrown and the sub-resources need to be deleted first.

Reply:
To update the information, the implementation in kv store is also implemented in the upper layer. If `cascade` is set to `true`, cascade deletion will occur, so whether to cascade deletion depends on the logic of the upper layer. I will add a `TODO` to implement in subsequent PRs. If `cascade` is `false`, check if the the sub-resources are empty, if it is not empty, deletion is not allowed.

Reply:
Done